core/app/models/workarea/content/block.rb

```diff
@@ -39,6 +39,11 @@ class Block
       #
       field :hidden_breakpoints, type: Array, default: []
 
+      # @!attribute custom_name
+      #   @return [String] the user-defined name of the block
+      #
+      field :custom_name, type: String, default: nil
```

Comment:
default is always nil if not defined, no need to be explicit. Think the field should just be `name` and then you can define the getter

```
def name
  super || BlockName.new(self).to_s
end
```